### PR TITLE
Move the error description to a separate allocation as it is rarely u…

### PR DESCRIPTION
**Pull request by baldersheim**
…sed.

This reduces the size of frequently used objects.

@geirst PR

### CODE REVIEW

**Review by geirst — APPROVED** [review]
👍  @vekterli FYI

File: storageapi/src/vespa/storageapi/messageapi/returncode.cpp [hunk 1]
```diff
@@ -145,5 +167,21 @@ ReturnCode::isBucketDisappearance() const
             return false;
     }
 }
+namespace {
+    vespalib::stringref Empty("");
+}
+vespalib::stringref
+ReturnCode::getMessage() const {
+    return _message ? _message->operator vespalib::stringref() : Empty;
```
Comment (vekterli) [inline storageapi/src/vespa/storageapi/messageapi/returncode.cpp:175]: Is the global object really needed here, vs. just returning a default-constructed `stringref`?

**Review by vekterli — COMMENTED** [review]
:+1:

File: storageapi/src/vespa/storageapi/messageapi/returncode.cpp (hunk 1 above)
Comment (baldersheim) [reply storageapi/src/vespa/storageapi/messageapi/returncode.cpp:175]: No, you are right. Old habit....

**Pull request merged**